### PR TITLE
Add dev rockspec and CI workflows for lint & deploy

### LINKED ISSUE #2: Set up CI workflows
@alerque let me know if there are any prerequisites not yet met. I note that we don't have any luarocks packaging yet, but there is a luacheckrc.

**Update:** I've added a proposed rockspec at https://github.com/lunarmodules/lua-lace/tree/feature/add-rockspec so feel free to merge that branch if it looks good enough.

### PR DESCRIPTION
**Pull request by alerque**
Closes #2

### CODE REVIEW

**Comment by alerque** [comment]
This is just a bit tricky to test end to end without merging. These actions will not run at all from this PR, they have to exist in the default branch at least once before they will run at all.

I tested on my fork and at least the lint and rockspec build tests pass, you can see [the output of a test run here](https://github.com/alerque/lua-lace/actions/runs/2552013311). It didn't attempt to upload because it was from my fork not the canonical repository.

Note I applied the rockspec commit you mentioned *after* the CI workflow one because the workflows will detect if the rockspec is touched and only attempt to test and publish it if it was modified.

File: lua-lace-dev-1.rockspec [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package = "lua-lace"
+version = "dev-1"
+source = {
+   url = "git+ssh://git@github.com/lunarmodules/lua-lace.git"
```
Comment (alerque) [inline lua-lace-dev-1.rockspec]:
```suggestion
   url = "git+https://git@github.com/lunarmodules/lua-lace.git"
```

**Review by alerque — COMMENTED** [review]
Default access should be over unauthenticated protocols, not everybody will have their SSH keys in GitHub to fetch this way.

File: lua-lace-dev-1.rockspec [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package = "lua-lace"
+version = "dev-1"
+source = {
+   url = "git+https://git@github.com/lunarmodules/lua-lace.git"
```
Comment (kinnison) [inline lua-lace-dev-1.rockspec]: If you're going to use http you don't need the username
```suggestion
   url = "git+https://github.com/lunarmodules/lua-lace.git"
```

Comment (mwild1) [reply lua-lace-dev-1.rockspec]: We might want to go a step further, and remove the `git+` prefix too. The official URL that Github provides for unauthenticated pulls is `https://github.com/lunarmodules/lua-lace.git`

Comment (alerque) [reply lua-lace-dev-1.rockspec]: Oops, yes that should go too. I'll amend...

**Review by mwild1 — APPROVED** [review]

**Comment by mwild1** [comment]
Okay, it all worked... except my last suggestion was a bad one :)

It seems that *luarocks* uses the `git+` prefix to determine that it's a Git repository to clone. Without that prefix, it just downloads a HTML page and tries to extract it as an archive. I'll push a fix.

**Comment by alerque** [comment]
Oops. Sorry I actually should have known that... you should be able to push a fix directly to *master*, any commits that touch the dev rockspec on the *master* branch should cause a force push (force push does not apply to tagged release rockspecs).

**Pull request merged**